addons/web_editor/models/html_field_history_mixin.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+from .diff_utils import apply_patch, generate_comparison, generate_patch
+
+
+class HtmlFieldHistory(models.AbstractModel):
+    _name = "html.field.history.mixin"
+    _description = "Field html History"
+    _max_html_field_history_revisions = 300
+
+    html_field_history_revisions = fields.Json(
+        "History revisions data", prefetch=False
+    )
+
+    html_field_history_metadata = fields.Json(
+        "History revisions metadata", compute="_compute_metadata", store=True
+    )
```

Comment:
@odony I discussed with Antony at lunch, he thinks this field is not very useful because as soon as you'll want to display a comparison with a previous version, you'll have to query the database for the full `html_field_history_revisions` anyway. In this regard, we thought it might not be worth the effort but, even if it seems useless from a database query point of view, its impact might be more interesting from a network rpc point of view. Without it, you'd need to download all the patches just to display the history dialog. What is your opinion on this ?